src/ucs/sys/topo/base/topo.c

```diff
@@ -306,6 +341,53 @@ ucs_topo_get_distance_sysfs(ucs_sys_device_t device1,
     return UCS_OK;
 }
 
+static ucs_status_t
+ucs_topo_get_avg_distance_sysfs(ucs_sys_device_t device,
+                                ucs_sys_dev_distance_t *distance)
+{
+    double total_distance = 0;
+    ucs_sys_cpuset_t process_affinity;
+    int cpu, num_cpus;
+    char path[PATH_MAX];
+    ucs_numa_node_t dev_node;
+    ucs_status_t status;
+    int ret;
+
+    /* If the device is unknown, we assume min distance */
+    if (device == UCS_SYS_DEVICE_ID_UNKNOWN) {
+        return ucs_topo_get_avg_distance_default(device, distance);
+    }
+
+    ret = ucs_sys_getaffinity(&process_affinity);
```

Comment:
Need to use thread affinity instead (`pthread_getaffinity_np`).
Please add a func in `ucs_sys`, reuse in `uct_ib_md_handle_mr_list_multithreaded`.

Reply:
@ofirfarjun7 unresoved?